packages/core/sdk/__tests__/auth/internal/CoreStitchAuthUnitTests.ts

```diff
@@ -248,46 +253,285 @@ describe("CoreStitchAuthUnitTests", () => {
       });
   });
 
-  it("should logout", () => {
+  it("should logout", async () => {
     const requestClientMock = getMockedRequestClient();
     const requestClient = instance(requestClientMock);
     const routes = new StitchAppRoutes("my_app-12345").authRoutes;
+    const storage = new MemoryStorage(appId);
     const auth = new StitchAuth(
       requestClient,
       routes,
-      new MemoryStorage(appId)
+      storage
     );
 
     expect(auth.isLoggedIn).toBeFalsy();
+    mockNextUserResponse(requestClientMock);
+    const user1 = await auth.loginWithCredentialInternal(
+      new AnonymousCredential()
+    );
 
-    return auth
-      .loginWithCredentialInternal(new AnonymousCredential())
-      .then(() => {
-        expect(auth.isLoggedIn).toBeTruthy();
+    mockNextUserResponse(requestClientMock);
+    const user2 = await auth.loginWithCredentialInternal(
+      new UserPasswordCredential("hi", "there")
+    );
 
-        return auth.logoutInternal();
-      })
-      .then(() => {
-        verify(requestClientMock.doRequest(anything())).times(3);
+    mockNextUserResponse(requestClientMock);
+    const user3 = await auth.loginWithCredentialInternal(
+      new UserPasswordCredential("bye", "there")
+    );
 
-        const expectedRequest = new StitchRequest.Builder();
-        expectedRequest.withMethod(Method.DELETE).withPath(routes.sessionRoute);
-        const headers = {
-          [Headers.AUTHORIZATION]: Headers.getAuthorizationBearer(
-            TEST_REFRESH_TOKEN
-          )
-        };
-        expectedRequest.withHeaders(headers);
+    expect(auth.listUsers()[2]).toEqual(user3);
+    expect(auth.user).toEqual(user3);
+    expect(auth.isLoggedIn).toBeTruthy();
 
-        const [actualRequest] = capture(
-          requestClientMock.doRequest
-        ).byCallIndex(2);
-        expect(expectedRequest.build()).toEqualRequest(actualRequest);
+    await auth.logoutInternal();
 
-        expect(auth.isLoggedIn).toBeFalsy();
-      });
+    expect(auth.isLoggedIn).toBeFalsy();
```

Comment:
I would try testing how some of these methods fail. Like how calling switch/logout with a non-existent user id should throw a userNotFound

Reply:
I test the switch case in the switch test but I'll add the logout case